src/main/java/Parser.java

```diff
@@ -0,0 +1,102 @@
+import java.io.IOException;
+
+public class Parser {
+    private static final String SAVED_FORMAT_DELIMITER = " \\| ";
+    private final TaskList taskList;
+
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Parse input from file and return the corresponding task
+     * @param taskString
+     * @return Task
+     */
+
+    public static Task parseFileInput(String taskString) {
+        String[] split = taskString.split(SAVED_FORMAT_DELIMITER);
+        String type = split[0];
+        String isDone = split[1];
+        switch (type) {
+            case "T":
+                String descriptionToDo = split[2];
+                ToDo taskT = new ToDo(descriptionToDo);
+                if (isDone.equals("X")) {
+                    taskT.markAsDone();
+                }
+                return taskT;
+            case "D":
+                String descriptionDeadlines = split[2];
+                String byDeadlines = split[3];
+                Deadlines taskD = new Deadlines(descriptionDeadlines, byDeadlines, true);
+                if (isDone.equals("X")) {
+                    taskD.markAsDone();
+                }
+                return taskD;
+            case "E":
+                String descriptionEvents = split[2];
+                String byEvents = split[3];
+                Deadlines taskE = new Deadlines(descriptionEvents, byEvents, true);
+                if (isDone.equals("X")) {
+                    taskE.markAsDone();
+                }
+                return taskE;
+            default:
+                return null;
+
+        }
+
+    }
+
+    /**
+     * Parse input from user and add the task to taskList
+     * @param commandFromUser
+     */
+    public String parseUserCommand(String commandFromUser) throws DukeException {
```

Comment:
Consider isolating the execution of different commands into their individual Command classes